effect/shared/src/main/scala/scalaz/effect/Async.scala

```diff
@@ -32,7 +31,8 @@ object Async {
    *
    * See `IO.async0` for more information.
    */
-  final def later[E, A]: Async[E, A] = Later()
+  final def later[E, A]: Async[E, A] =
+    MaybeLater(NoOpCanceler)
 
```

Comment:
I am sorry, we have to make one more change here, to eliminate the overhead of this common case:

```scala
final def later[E, A]: Async[E, A] = _Later.asInstanceOf[Async[E, A]]

private val _Later: Async[Nothing, Nothing] = MaybeLater(NoOpCanceler)
```

Otherwise, we have to incur the overhead of allocating `MaybeLater` every time `later` is called, which will noticeably affect performance.

Reply:
👍  I'm doing it. Thanks

Reply:
I was wondering if it shouldn't be done also for `maybeLater`, `maybeLaterIO` and `now` ?